lib/HLSL/DxilPreparePasses.cpp

```diff
@@ -446,6 +449,22 @@ class DxilFinalizeModule : public ModulePass {
   }
 
 private:
+  void ConsolidateAllocas(Module &M) {
+    for (Function &F : M) {
+      if (F.isDeclaration())
+        continue;
+      Instruction *insertPt = nullptr;
+      for (llvm::Instruction &I : llvm::inst_range(&F)) {
+        if (!insertPt) {
+          if (!isa<AllocaInst>(I) && !isa<DbgInfoIntrinsic>(I))
+            insertPt = &I;
+        } else if (isa<AllocaInst>(I)) {
+          I.moveBefore(insertPt);
+        }
+      }
+    }
+  }
+
```

Comment:
Besides the simple removal of SkipAllocas, this is the meat of the change. It is required because validation will find that uses of the allocas are not dominated by them if they are left where they are inserted.

It does change the output a bit. Previously, loads of globals and inputs ended up before the allocas. This alteration necessitates the minor test changes.